tls/extensions/s2n_cert_authorities.c

```diff
@@ -14,8 +14,79 @@
  */
 #include "tls/extensions/s2n_cert_authorities.h"
 
+#include <openssl/x509.h>
+
 #include "utils/s2n_safety.h"
 
+bool s2n_cert_authorities_supported_from_trust_store()
+{
+#if S2N_LIBCRYPTO_SUPPORTS_X509_STORE_LIST
+    return true;
+#else
+    return false;
+#endif
+}
+
+static S2N_RESULT s2n_cert_authorities_set_from_trust_store(struct s2n_config *config, uint16_t *count)
+{
+    RESULT_ENSURE_REF(config);
+    RESULT_ENSURE_REF(count);
+    *count = 0;
+
+    if (!config->trust_store.trust_store) {
+        return S2N_RESULT_OK;
+    }
+
+#if S2N_LIBCRYPTO_SUPPORTS_X509_STORE_LIST
```

Comment:
Nit: is there a reason to use an if/def rather than the `s2n_cert_authorities_supported_from_trust_store` helper?

Reply:
We need conditional compilation. If we used s2n_cert_authorities_supported_from_trust_store, then we'd get a compile-time error for the missing methods.